### ISSUE TITLE
Suggestion: change createToken(too, many, params) into createToken(params_object)

### ISSUE THREAD

**Issue by Richie765** [kyndryl-open-source/hashi-vault-js#6]
https://github.com/rod4n4m1/hashi-vault-js/blob/6ecc8a1e0ea6da57a0afc7cbe5aeb021b7986ce6/Vault.js#L281-L283

Functions with more than a few arguments become very impractical. Its better to do something like this:

```javscript
async createToken(vaultToken, params) {
  params = {
    // defaults, for instance
    type: 'service', 
    // overwrite defaults with user suppied
    ...params
  };

  const { id, role_name, policies, meta, no_parent,
    no_default_policy, renewable, ttl, type, explicit_max_ttl, display_name,
    num_uses, period, entity_alias } = params;

  // rest of the code
}
```

As you can see, I changed the names of the parameters to match exactly the names as they are with the Vault API (e.g. roleName => role_name). I think that makes it easier to use overall if the two match up.

I also kept the vaultToken out of the params object as its functionally different from the rets.

I also see this function has the same mistake with the `typeof == undefined`.

Also, is there a reason why the helpers like `createSToken` pass `calledFunction` in the metadata? I don't think this is generally desired, so it should be up to the user to do so themselves if needed.

If you are interested I could look deeper into the code and see what other suggestions I could make.

**Comment by rod-anami** [comment]
Thanks again @Richie765! That passed through my mind as well. I'll address this refactoring on `0.3.20`.

**Comment by Richie765** [comment]
Great, thanks! Keep in mind this is a breaking change, so when you publish to npm it should become 0.4.x. Also I believe the  `createSToken` and `createBToken` become kind of obsolete since the same can be easily accomplished with the new `createToken`.

**Comment by rod-anami** [comment]
That's true! Thanks for reminding me about npm versioning scheme. I tagged this one with `0.4.0`.

**Comment by Richie765** [comment]
Awesome! Looking forward to it. I'm not sure if you noticed but #5 can be reopened. Thanks for putting in your effort!

**Comment by Richie765** [comment]
Would you like I make a PR for this or you prefer to do it yourself?

**Comment by rod-anami** [comment]
@Richie765 It'd be nice if you can contribute directly in the code, let me create a new branch for the `0.4.0` version, so you can do a PR for it. Don't worry about documentation in the readme and changelog, I can take care of that.

**Comment by Richie765** [comment]
@rod4n4m1, great, I'm almost finished with it.

Speaking of the changelog, I think it is good to move it from /docs/CHANGELOG.md to the project root and it is not really needed in the readme. Usually when something stops working after an update, this is the first I'm looking for. This is common in many projects as well.

**Comment by rod-anami** [comment]
Merged from `0.4.0` branch to master.
